lber: v0.4.0 doesn't compile on armv7

I updated to lber v0.4.0 (by updating ldap3_proto), but it seems to break the armv7-unknown-linux-gnueabihf build:

```
error[E0277]: the trait bound `u64: ToUsize` is not satisfied
   --> /__w/lldap/lldap/${GITHUB_WORKSPACE}/.cargo/registry/src/github.com-1ecc6299db9ec823/lber-0.4.0/src/parse.rs:52:37
    |
52  |             let (j, content) = take(len)(i)?;
    |                                ---- ^^^ the trait `ToUsize` is not implemented for `u64`
    |                                |
    |                                required by a bound introduced by this call
    |
    = help: the following other types implement trait `ToUsize`:
              u16
              u32
              u8
              usize
note: required by a bound in `nom::bytes::streaming::take`
   --> /__w/lldap/lldap/${GITHUB_WORKSPACE}/.cargo/registry/src/github.com-1ecc6299db9ec823/nom-7.1.3/src/bytes/streaming.rs:416:6
    |
416 |   C: ToUsize,
    |      ^^^^^^^ required by this bound in `nom::bytes::streaming::take`

error[E0277]: the trait bound `u64: ToUsize` is not satisfied
   --> /__w/lldap/lldap/${GITHUB_WORKSPACE}/.cargo/registry/src/github.com-1ecc6299db9ec823/lber-0.4.0/src/parse.rs:58:41
    |
58  |             let (j, mut content) = take(len)(i)?;
    |                                    ---- ^^^ the trait `ToUsize` is not implemented for `u64`
    |                                    |
    |                                    required by a bound introduced by this call
    |
    = help: the following other types implement trait `ToUsize`:
              u16
              u32
              u8
              usize
note: required by a bound in `nom::bytes::streaming::take`
   --> /__w/lldap/lldap/${GITHUB_WORKSPACE}/.cargo/registry/src/github.com-1ecc6299db9ec823/nom-7.1.3/src/bytes/streaming.rs:416:6
    |
416 |   C: ToUsize,
    |      ^^^^^^^ required by this bound in `nom::bytes::streaming::take`
```

That's probably because the platform is 32 bits, so going from u64 to usize (32 bits) isn't supported.

--- a/lber/src/parse.rs
+++ b/lber/src/parse.rs
@@ -1,3 +1,5 @@
+use std::convert::TryFrom;
+
 use common::TagClass;
 use common::TagStructure;
 use structure::{StructureTag, PL};
@@ -6,6 +8,7 @@ use nom;
 use nom::bits::streaming as bits;
 use nom::bytes::streaming::take;
 use nom::combinator::map_opt;
+use nom::error::{Error, ErrorKind, ParseError};
 use nom::number::streaming as number;
 use nom::sequence::tuple;
 use nom::{IResult, InputLength, Needed};
@@ -26,15 +29,19 @@ fn parse_type_header(i: &[u8]) -> nom::IResult<&[u8], (TagClass, TagStructure, u
     nom::bits(tuple((class_bits, pc_bit, tagnr_bits)))(i)
 }
 
-fn parse_length(i: &[u8]) -> nom::IResult<&[u8], u64> {
+fn parse_length(i: &[u8]) -> nom::IResult<&[u8], usize> {
     let (i, len) = number::be_u8(i)?;
     if len < 128 {
-        Ok((i, len as u64))
+        Ok((i, len as usize))
     } else {
         let len = len - 128;
         let (i, b) = take(len)(i)?;
         let (_, len) = parse_uint(b)?;
-        Ok((i, len))
+        Ok((
+            i,
+            usize::try_from(len)
+                .map_err(|_| nom::Err::Failure(Error::from_error_kind(i, ErrorKind::TooLarge)))?,
+        ))
     }
 }
 

